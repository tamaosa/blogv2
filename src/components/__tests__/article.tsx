import React from "react"
import { useStaticQuery } from "gatsby"
import { render, screen } from "@testing-library/react"
import Article from "../article"

describe("Article", () => {
  const post = (slug: string) => {
    return {
      id: "id",
      excerpt: "excerpt",
      body: "body",
      frontmatter: {
        title: "title",
        published: "2022-01-01",
        updated: "2022-12-31",
        tags: ["tag1", "tag2", "tag3", "tag4", "tag5"],
      },
      fields: {
        slug: slug,
      },
    }
  }

  const repository = "repository"

  beforeAll(() => {
    ;(useStaticQuery as jest.Mock).mockReturnValue({
      site: {
        siteMetadata: {
          repository: repository,
        },
      },
    })
  })

  afterAll(() => {
    jest.clearAllMocks()
  })

  it("snapshot test", () => {
    const { asFragment } = render(
      <Article
        post={post("/slug1/")}
        relatedPosts={[post("/slug2/")]}
        prevPost={post("/slug3/")}
        nextPost={post("/slug4/")}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it("testing title", () => {
    const { container } = render(
      <Article
        post={post("/slug1/")}
        relatedPosts={[post("/slug2/")]}
        prevPost={post("/slug3/")}
        nextPost={post("/slug4/")}
      />
    )
    expect(container.innerHTML).toMatch(`<h1 itemprop="headline">title</h1>`)
  })

  it("testing link to github", () => {
    const { container } = render(
      <Article
        post={post("/slug1/")}
        relatedPosts={[post("/slug2/")]}
        prevPost={post("/slug3/")}
        nextPost={post("/slug4/")}
      />
    )
    expect(container.innerHTML).toMatch(
      `<a target="_blank" href="repository/tree/master/content/entries/slug1/index.mdx" rel="external noopener">（GitHubで編集を提案）</a>`
    )
  })

  it("testing no related post", () => {
    const { container } = render(
      <Article
        post={post("/slug1/")}
        relatedPosts={[]}
        prevPost={post("/slug2/")}
        nextPost={post("/slug3/")}
      />
    )
    expect(container).not.toHaveTextContent("関連記事")
    const posts = screen.queryAllByTestId("article-link")
    expect(posts.length).toEqual(0)
  })

  it("testing related post", () => {
    const { container } = render(
      <Article
        post={post("/slug1/")}
        relatedPosts={[post("/slug2/"), post("/slug3/"), post("/slug4/")]}
        prevPost={post("/slug5/")}
        nextPost={post("/slug6/")}
      />
    )
    expect(container).toHaveTextContent("関連記事")
    const posts = screen.queryAllByTestId("article-link")
    expect(posts.length).toEqual(3)
  })

  it("testing link to prevPost", () => {
    const { container } = render(
      <Article
        post={post("/slug1/")}
        relatedPosts={[post("/slug2/")]}
        prevPost={post("/slug3/")}
        nextPost={post("/slug4/")}
      />
    )
    expect(container.innerHTML).toMatch(
      `<a rel="prev" href="/slug3/">← title</a>`
    )
  })

  it("testing link to nextPost", () => {
    const { container } = render(
      <Article
        post={post("/slug1/")}
        relatedPosts={[post("/slug2/")]}
        prevPost={post("/slug3/")}
        nextPost={post("/slug4/")}
      />
    )
    expect(container.innerHTML).toMatch(
      `<a rel="next" href="/slug4/">title →</a>`
    )
  })
})
