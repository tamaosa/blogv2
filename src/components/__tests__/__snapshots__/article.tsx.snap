// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article snapshot test 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin-bottom: 1.5rem;
}

.emotion-0 div {
  text-align: right;
  margin-top: 0.75rem;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row wrap;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  list-style: none;
  margin: 0;
}

.emotion-1 li {
  margin: 0 0.375rem 0 0;
}

.emotion-1 a {
  color: var(--fg);
  border: solid 2px var(--fg);
  border-radius: 10px;
  padding: 0.1875rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1 a:hover {
  color: var(--bg);
  background: var(--fg-title);
  border: solid 2px var(--fg-title);
}

.emotion-2 {
  display: inline-block;
  margin-right: 0.375rem;
}

.emotion-4 {
  margin-bottom: 1.5rem;
}

.emotion-4 div:last-child {
  text-align: right;
  margin-bottom: 1.5rem;
}

.emotion-4 div:last-child a {
  font-size: small;
}

.emotion-5 {
  display: block;
  text-align: center;
  margin-bottom: 0.75rem;
}

.emotion-6 {
  text-align: center;
  font-weight: bold;
  font-size: 1.18921rem;
  margin-bottom: 0.375rem;
}

.emotion-7 {
  font-size: 1.29684rem;
  font-weight: bold;
  line-height: 1;
}

.emotion-7 a {
  color: var(--fg-title);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row wrap;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 0.8409rem;
  margin: 0.375rem 0 0.75rem 0;
}

.emotion-12 ul {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row wrap;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  list-style: none;
  margin: 0;
  padding: 0;
}

.emotion-12 li:nth-of-type(2n) {
  margin-left: auto;
}

<article
    itemscope=""
    itemtype="http://schema.org/Article"
  >
    <div
      class="emotion-0"
    >
      <h1
        itemprop="headline"
      >
        title
      </h1>
      <div>
        <ul
          class="emotion-1"
        >
          <li>
            <a
              aria-label="タグへのリンク"
              href="/tags/tag1/"
            >
              #tag1
            </a>
          </li>
          <li>
            <a
              aria-label="タグへのリンク"
              href="/tags/tag2/"
            >
              #tag2
            </a>
          </li>
          <li>
            <a
              aria-label="タグへのリンク"
              href="/tags/tag3/"
            >
              #tag3
            </a>
          </li>
          <li>
            <a
              aria-label="タグへのリンク"
              href="/tags/tag4/"
            >
              #tag4
            </a>
          </li>
          <li>
            <a
              aria-label="タグへのリンク"
              href="/tags/tag5/"
            >
              #tag5
            </a>
          </li>
        </ul>
      </div>
      <div>
        <span
          class="emotion-2"
        >
          <time>
            2022年1月1日
          </time>
           公開
        </span>
        <span
          class="emotion-2"
        >
           
          <time>
            2022年12月31日
          </time>
           更新
        </span>
      </div>
    </div>
    <div
      class="emotion-4"
    >
      <div>
        body
      </div>
      <div>
        <a
          href="repository/tree/master/content/entries/slug1/index.mdx"
          rel="external noopener"
          target="_blank"
        >
          （GitHubで編集を提案）
        </a>
      </div>
    </div>
    <aside>
      <ins
        class="adsbygoogle emotion-5"
        data-ad-format="auto"
        data-full-width-responsive="true"
      />
      <div>
        <div
          class="emotion-6"
        >
          <span>
            関連記事
          </span>
        </div>
        <ol
          style="list-style: none;"
        >
          <li
            data-testid="article-link"
          >
            <article>
              <div
                class="emotion-7"
              >
                <a
                  aria-label="記事へのリンク"
                  href="/slug2/"
                >
                  title
                </a>
              </div>
              <div
                class="emotion-8"
              >
                <div>
                  <span
                    class="emotion-2"
                  >
                    <time>
                      2022年1月1日
                    </time>
                     公開
                  </span>
                  <span
                    class="emotion-2"
                  >
                     
                    <time>
                      2022年12月31日
                    </time>
                     更新
                  </span>
                </div>
                <div>
                  <div>
                    <ul
                      class="emotion-1"
                    >
                      <li>
                        <a
                          aria-label="タグへのリンク"
                          href="/tags/tag1/"
                        >
                          #tag1
                        </a>
                      </li>
                      <li>
                        <a
                          aria-label="タグへのリンク"
                          href="/tags/tag2/"
                        >
                          #tag2
                        </a>
                      </li>
                      <li>
                        <a
                          aria-label="タグへのリンク"
                          href="/tags/tag3/"
                        >
                          #tag3
                        </a>
                      </li>
                      <li>
                        <a
                          aria-label="タグへのリンク"
                          href="/tags/tag4/"
                        >
                          #tag4
                        </a>
                      </li>
                      <li>
                        <a
                          aria-label="タグへのリンク"
                          href="/tags/tag5/"
                        >
                          #tag5
                        </a>
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
            </article>
          </li>
        </ol>
      </div>
      <div
        class="emotion-12"
      >
        <ul>
          <li>
            <a
              href="/slug3/"
              rel="prev"
            >
              ← title
            </a>
          </li>
          <li>
            <a
              href="/slug4/"
              rel="next"
            >
              title →
            </a>
          </li>
        </ul>
      </div>
    </aside>
  </article>
</DocumentFragment>
`;
